cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/Plugin.java

```diff
@@ -48,9 +48,15 @@ public String getName() {
     return properties;
   }
 
-  @Nullable
-  public ArtifactId getArtifactId() {
-    return artifactId;
+  /**
+   * @param pluginType the plugin type to get a selector for. Only used for error messages
```

Comment:
``error messages when we are trying to select an artifact for plugin`` would be better right ?